dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50Coercion.war/src/io/openliberty/el50/fat/coercion/servlets/lambda/EL50LambdaExpressionCoercionServlet.java

```diff
@@ -0,0 +1,118 @@
+ */
+@WebServlet({ "/EL50LambdaExpressionCoercion" })
+public class EL50LambdaExpressionCoercionServlet extends FATServlet {
+    private static final long serialVersionUID = 1L;
+
+    ELProcessor elp;
+
+    public EL50LambdaExpressionCoercionServlet() {
+        super();
+
+        elp = new ELProcessor();
+        elp.defineBean("lambdaBean", new LambdaBean());
+    }
+
+    @Test
+    public void testLambdaCoercion() throws Exception {
+        // Test lambda coercion basic test passing a lambda as a parameter from a defined bean
+        String result;
+        String expectedResult ="success";
+        Object obj = elp.eval("lambdaBean.testLambdaCoercion(param -> true)");
+        assertNotNull(obj);
+        result = obj.toString();
+
+        assertEquals("The expression did not evaluate to: " + expectedResult + " but was: " + result, expectedResult, result);
+    }
+
+    @Test
+    public void testLambdaCoercionWrongType() throws Exception {
+        // Test lambda with a wrong type by passing a lambda evaluating with a wrong type
+        Object result = null;
+        boolean exceptionCaught = false;
+        try {
+            result = elp.eval("lambdaBean.testIntLambdaCoercion(param -> param.equals('should be int not string'))");
+        } catch (ELException e) {
+            exceptionCaught = true;
+        }
+        assertEquals("The expression did not throw expected ELException. Got result: "+result, true, exceptionCaught);
+    }
+
+    @Test
+    public void testLambdaCoercionMissingLambda() throws Exception {
+        // Test lambda coercion by passing a lambda that is not defined
+        Object result = null;
+        boolean exceptionCaught = false;
+        try {
+            result = elp.eval("lambdaBean.missingLambda(param -> true)");
+        } catch (ELException e) {
+            exceptionCaught = true;
+        }
+        assertEquals("The expression did not throw expected ELException. Got result: "+result, true, exceptionCaught);
+    }
+
+    @Test
+    public void testLambdaCoercionNotALambda() throws Exception {
+        // Test by not passing a lambda as parameter but a normal string
+        Object result = null;
+        boolean exceptionCaught = false;
+        try {
+            result = elp.eval("lambdaBean.testLambdaCoercion('noLambdaPassed')");
+        } catch (ELException e) {
+            exceptionCaught = true;
+        }
+        assertEquals("The expression did not throw expected ELException. Got result: "+result, true, exceptionCaught);
+    }
+
+    public class LambdaBean {
+
+        public String testLambdaCoercion(Predicate<String> filter) {
+            return "success";
+        }
+
+        public String testIntLambdaCoercion(Predicate<Integer> filter) {
+            // Expected string to test against
+            Integer expectedInteger = 184;
+            if(filter.test(expectedInteger)){
+                return "success";
+            }
+            return "failed";
+        }
```

Comment:
Do you have a test case where testIntLambdaCoercion returns success?  The `testLambdaCoercion` method I created doesn't call the test function.

Reply:
Doesn't exactly return success, the test case where it is used requires the lambda to be evaluated in order to assess the failure that's why the method class to test and expects a failure. I made it return success of failed in case future tests are added and can reuse the bean but it isn't really necessary we just need to evaluate it.